internal/action/actions.go

```diff
@@ -172,9 +172,20 @@ func (h *BufPane) MoveCursorUp(n int) {
 			h.Cursor.Loc = h.Buf.Start()
 			h.Cursor.LastVisualX = 0
 		} else {
+			// store the row the cursor is moving from
+			movingfromrow := vloc.SLoc.Row
 			vloc.SLoc = sloc
 			vloc.VisualX = h.Cursor.LastVisualX
 			h.Cursor.Loc = h.LocFromVLoc(vloc)
+			if util.IntOpt(h.Buf.Settings["wrapindent"]) > -1 {
+				// Reset cursor's X position if we moved up from row 0 to another paragraph
+				if movingfromrow == 0 {
```

Comment:
This is not behaving as it should at the moment. This should have the same logic as `MoveCursorDown`, no?

i.e. If we are moving **to** (not from) row 0, `VisualX` should be 0 otherwise we adjust the wrap indent with `AdjustXCursorOnWrapindent`

Reply:
Thanks for the tip, have refactored and simplified the codebase a bit so it now works as it should.